src/core/src/ffi/signature.rs

```diff
@@ -165,11 +169,17 @@ ffi_fn! {
 unsafe fn signature_first_mh(ptr: *const SourmashSignature) -> Result<*mut SourmashKmerMinHash> {
     let sig = SourmashSignature::as_rust(ptr);
 
-    if let Some(Sketch::MinHash(mh)) = sig.signatures.get(0) {
-        Ok(SourmashKmerMinHash::from_rust(mh.clone()))
-    } else {
-        // TODO: need to select the correct one
-        unimplemented!()
+    match sig.signatures.get(0) {
+        Some(Sketch::MinHash(mh)) => {
+            Ok(SourmashKmerMinHash::from_rust(mh.clone()))
+        },
+        Some(Sketch::LargeMinHash(mh_btree)) => {
+            let mh = KmerMinHash::from(mh_btree);
+            Ok(SourmashKmerMinHash::from_rust(mh))
```

Comment:
Note that once you implement `From` you get `Into` for free:
```suggestion
        Some(Sketch::LargeMinHash(mh)) => {
            Ok(SourmashKmerMinHash::from_rust(mh.into()))
```
and you also don't need to `use crate::sketch::minhash::KmerMinHash;`

Reply:
🤩

Reply:
addressed in 9d536d1f